include/radix/Window.hpp

```diff
@@ -197,13 +247,20 @@ class Window : public Viewport, public InputSource {
   void initGl();
 
 
-  unsigned int  width;          /**< main screen width */
-  unsigned int  height;         /**< main screen height */
-  SDL_Window   *window;         /**< SDL identifier for current window */
-  SDL_GLContext context;        /**< SDL Handler for OpenGL Context */
+  unsigned int  width;                      /**< main screen width */
+  unsigned int  height;                     /**< main screen height */
+  SDL_Window   *window;                     /**< SDL identifier for current window */
+  SDL_GLContext context;                    /**< SDL Handler for OpenGL Context */
+
+  SDL_Joystick *joystick;
+  SDL_GameController* controller;
 
-  std::vector<bool> keystates;  /**< Keyboard pressed key status */
-  std::string       charbuffer; /**< Text input buffer */
+  std::vector<bool> keystates;              /**< Keyboard key pressed status */
+  std::vector<bool> controllerbuttonstates; /**< Controller button pressed status */
+  std::vector<bool> mousebuttonstates;  /**< Mouse button pressed status */
+  int mouse_xrel;                           /**< Mouse relative x movement */
+  int mouse_yrel;                           /**< Mouse relative y movement */
```

Comment:
These should be camel case and you can get rid of the the mnemonic. mouseRelativeY, mouseRelativeX for example.